src/expr.rs

```diff
@@ -2101,6 +2101,25 @@ pub(crate) fn rewrite_assign_rhs_expr<R: Rewrite>(
     rhs_kind: &RhsAssignKind<'_>,
     rhs_tactics: RhsTactics,
 ) -> Option<String> {
+    let get_lhs_last_line_shape = || {
+        let last_line = lhs.rsplitn(2, "\n").next().unwrap_or_default();
+        let tab_spaces = context.config.tab_spaces();
+        let new_shape = shape
+            .block_indent(tab_spaces)
+            .saturating_sub_width(tab_spaces);
+        let extra_indent_string = new_shape.to_string(&context.config).to_string();
+        if last_line.starts_with(&extra_indent_string) {
+            new_shape
+        } else {
+            shape
+        }
+    };
+    let shape = if context.config.style_edition() >= StyleEdition::Edition2024 {
+        get_lhs_last_line_shape()
```

Comment:
I'm wondering if `get_lhs_last_line_shape` is the right name. Technically what we're doing here is getting the shape used to format the `rhs`, but it's based on the `lhs` indentation. If you think you can come up with a succinct name that conveys that meaning that would be great!

Reply:
in that case I will just call it get_rhs_shape, leave the implementation details out of the naming